pkg/client/automation/client.go

```diff
@@ -109,36 +111,47 @@ func WithClientRequestLimiter(limiter *concurrency.Limiter) func(client *Client)
 }
 
 // List returns all automation objects
-func (a Client) List(resourceType ResourceType) (result *ListResponse, err error) {
+func (a Client) List(resourceType ResourceType) (result *[]Response, err error) {
 	a.limiter.ExecuteBlocking(func() {
 		result, err = a.list(resourceType)
 	})
 	return
 }
 
-func (a Client) list(resourceType ResourceType) (*ListResponse, error) {
-	// try to get the list of resources
-	resp, err := rest.Get(a.client, a.url+a.resources[resourceType].Path)
-	if err != nil {
-		return nil, fmt.Errorf("unable to list automation resources: %w", err)
-	}
+func (a Client) list(resourceType ResourceType) (*[]Response, error) {
```

Comment:
does this still have to be a pointer? As a slice it can already be nil and the actual data of the underlying array is not copied if passed by value

Reply:
done